app/src/main/java/com/example/triptracker/MainActivity.kt

```diff
@@ -40,6 +42,8 @@ class MainActivity : ComponentActivity() {
           // Instance of NavController
           val navController = rememberNavController()
           val navigation = remember(navController) { Navigation(navController) }
+          val context = LocalContext.current
```

Comment:
The Application context needs to be global for all the screens of the app. This is why here the LocalContext.current can't be used. 
The better option is to create the global Application class that will handle this and create it only once. Here is an example : https://gist.github.com/paraya3636/bf8108a75eb49323e56c0c90dd0747e0